### ISSUE TITLE
Request For Comments: Set DEM Product Scaling Factor to 1

### ISSUE THREAD

**Issue by Saijin-Naib** [OpenDroneMap/ODM#1608]
### How did you install ODM? (Docker, installer, natively, ...)?
Docker, but this affects all platforms

### What is the problem?
Currently, we cap DEM products to 2x the GSD of the dataset.

### What should be the expected behavior? If this is a feature request, please describe in detail the changes you think should be made to the code, citing files and lines where changes should be made, if possible.
The main historical reasons for this have been to more accurately reflect the precision of our reconstruction, and to help keep resource consumption for this part of the pipeline under control.

I believe that at this point, with all the incredible work to reconstruction completeness, precision, accuracy, multi sub-resolution depthmaps, geometrically consistent depthmaps, rolling shutter correction, enhanced filtering, etc. that we no longer need to cap the DEM products to 2x dataset GSD, and that in fact, we are ready to have DEM products able to be created at 1:1.

### How can we reproduce this? What steps did you do to trigger the problem? If this is an issue with processing a dataset, YOU MUST include a copy of your dataset AND task output log, uploaded on Google Drive or Dropbox (otherwise we cannot reproduce this).
![image](https://user-images.githubusercontent.com/19295950/219909620-234859f9-962a-4805-8ff2-84771d85e113.png)
Brighton Beach Road - v3.0.4
```
Options:	crop: 0, dem-resolution: 0.1, dsm: true, dtm: true, feature-quality: ultra, min-num-features: 128000, orthophoto-resolution: 0.1, pc-filter: 0, pc-quality: ultra, primary-band: green, rolling-shutter: true
```

![image](https://user-images.githubusercontent.com/19295950/219909649-46a6b6d7-79e8-476f-a3a5-01ea5d594f66.png)
Brighton Beach Road - v3.0.4 1:1 GSD Patch
```
Options:	crop: 0, dem-resolution: 0.1, dsm: true, dtm: true, feature-quality: ultra, min-num-features: 128000, orthophoto-resolution: 0.1, pc-filter: 0, pc-quality: ultra, primary-band: green, rolling-shutter: true
```
Brighton Beach Road -v3.0.4
![image](https://user-images.githubusercontent.com/19295950/219909660-089d534c-dbbb-4a88-acb5-4428771eb78a.png)

I'm seeing a significant increase in real resolved detail with @smathermather patch.

![image](https://user-images.githubusercontent.com/19295950/219909806-3da27f4c-f485-499d-b916-4c2dbe061558.png)
ceitean-14793 -v3.0.4
```
Options:	crop: 0, dem-resolution: 1, dsm: true, dtm: true, min-num-features: 128000, orthophoto-resolution: 1, pc-filter: 0, pc-quality: high, primary-band: green, rerun-from: dataset, rolling-shutter: true
```
![image](https://user-images.githubusercontent.com/19295950/219909760-0a20d8a1-b2e9-4cd1-a204-1c158949eab9.png)
ceitean-14793 -v3.0.4 1:1 GSD Patch
```
Options:	crop: 0, dem-resolution: 1, dsm: true, dtm: true, min-num-features: 128000, orthophoto-resolution: 1, pc-filter: 0, pc-quality: high, primary-band: green, rerun-from: dataset, rolling-shutter: true
```
Resource consumption is slightly higher, but I've not fully profiled it.

I think that we can change our defaults for dem-resolution to 10cm/px to mimic the behavior of the current pipeline, while allowing folks who tune that parameter to get the full benefit of our improved processing pipeline.

**Comment by pierotofy** [comment]
Mm, if we increase the resolution, we need to make sure that it doesn't affect the radius. There was a reason the scaling factor was a multiple #914 and I think any benchmark should be done with defaults (e.g. no `pc-quality: ultra`, because most folks will not process with that).

**Comment by Saijin-Naib** [comment]
I'll redo a few more tomorrow on bog-standard Defaults and re-assess and post here. Thanks!

**Comment by smathermather** [comment]
> Mm, if we increase the resolution, we need to make sure that it doesn't affect the radius. There was a reason the scaling factor was a multiple #914 and I think any benchmark should be done with defaults (e.g. no `pc-quality: ultra`, because most folks will not process with that).

I wrote 914 and it was true for the time. Much like the closed source world, our point cloud is orders of magnitude better now, and I agree with @Saijin-Naib that the ceiling (but not the default) should reflect that. As such, the benchmark for quality improvements shouldn't be defaults. Nor should the only way to bypass this now when exceeding the defaults is with the unwise use of `ignore-gsd`.

A pull request to address this could be simple, i.e. remove gsd_scaling or set to 1.0, or be more sophisticated and change that value based on `pc-quality`.

**Comment by smathermather** [comment]
Simplest possible test here:
https://github.com/OpenDroneMap/ODM/pull/1609

Easy to refine and improve and does not change the default behavior (though we probably want to clean up "unused" variable in the form of a dem scaling factor.

**Comment by Saijin-Naib** [comment]
Settings for Old_Orchard:  
```
Options:  auto-boundary: true, dsm: true
```

---
v3.0.4:  

---
![image](https://user-images.githubusercontent.com/19295950/219965324-2f089c01-b8a0-4118-953a-43f93641aa8c.png)
![image](https://user-images.githubusercontent.com/19295950/219965685-5e8b9348-1d7c-41d6-9c7e-8ab983435789.png)
![image](https://user-images.githubusercontent.com/19295950/219965397-85c1165a-430b-4f1b-821b-5200b461db96.png)

---

v3.0.4 1:1 GSD Patch:  

---
![image](https://user-images.githubusercontent.com/19295950/219965540-483e0aaa-dc89-4837-9666-9b34a5ad574e.png)
![image](https://user-images.githubusercontent.com/19295950/219965442-4db32ad9-595f-4629-b029-1408d8ce5e0c.png)
![image](https://user-images.githubusercontent.com/19295950/219965492-515fda22-0e9d-4ded-a2db-a025d68c89cf.png)

---

PointCloud for Feature Reference:  
![image](https://user-images.githubusercontent.com/19295950/219965913-1d1cc5cc-8ac9-4531-8b77-3b346dd6376c.png)
![image](https://user-images.githubusercontent.com/19295950/219965805-e733d726-ee3a-4467-9a5a-0af1da78346d.png)
![image](https://user-images.githubusercontent.com/19295950/219965879-672774ef-3e15-4723-95a8-9cb27072a070.png)

**Comment by Saijin-Naib** [comment]
Settings for O'Brien's:  
```
Options:  auto-boundary: true, dsm: true
```

---
v3.0.4:  
![image](https://user-images.githubusercontent.com/19295950/219966063-178fd9ba-e2ea-497b-9a67-edcf2e5cab9b.png)
![image](https://user-images.githubusercontent.com/19295950/219966127-a23c05dc-2d98-4ff8-aa98-9ce3521e2198.png)
![image](https://user-images.githubusercontent.com/19295950/219966411-bc683a8d-9291-4b61-b3c9-2c7465d5930a.png)


---


---

v3.0.4 1:1 GSD Patch:  

---
![image](https://user-images.githubusercontent.com/19295950/219966090-ed2aa109-1761-4a93-8a5d-610be1db09d9.png)
![image](https://user-images.githubusercontent.com/19295950/219966137-3dd58b88-941f-478f-bb1f-ab099ab395d7.png)
![image](https://user-images.githubusercontent.com/19295950/219966424-c0ff8b98-e1b7-4f6a-aa7a-e0db5ec4fcab.png)


---

PointCloud for Feature Reference:  

---
![image](https://user-images.githubusercontent.com/19295950/219966506-cf93707c-deee-4946-9049-2fe27b0ca383.png)
![image](https://user-images.githubusercontent.com/19295950/219966613-408c7e8f-50c1-439d-b23c-cdde4fd7e1f7.png)
![image](https://user-images.githubusercontent.com/19295950/219966681-daf98652-f8f4-49e3-a2be-23895197d7ce.png)

**Comment by pierotofy** [comment]
I think I'm seeing more smoothing? That's probably due to the decrease in `dem-resolution` which consequently affects `radius_steps`, not necessarily because of GSD scaling.

**Comment by Saijin-Naib** [comment]
Yeah, on O'Brien's especially that seems to be the case with the bee-boxes on the far East of the field, but with Old_Orchard the 1:1 GSD patch was much better at picking up details in the field like the canopy footprints of the individual apple trees to the East, the drainage tile piping piles in the East, the various roof levels on the farmstead to the West, etc.

The positive improvement seems to echo what I observed with Brighton Beach as well, where separation between the planted islands and the roadway was improved and smaller features such as the stones lining the roadway and the wash-out channels in the drainage ditch being much more enhanced.

I'm going to see if I can make it on my Desktop when processing helenenschacht, given the significantly higher-quality data it is comprised of compared to my two datasets from Martin Farms (which were my first collects ever, so I made about every error and bad-practice possible).

I guess the question might become how do we tune radius_steps now?
